File: CS/MasterReport.cs
using DevExpress.DataAccess.Sql.DataApi;
using DevExpress.XtraReports.UI;
using System.Drawing.Printing;
using System.Windows.Forms;

namespace DrillThroughReport
{
    public partial class MasterReport : DevExpress.XtraReports.UI.XtraReport {
        public MasterReport() {
            InitializeComponent();
        }

        private void xrLabel1_BeforePrint(object sender, PrintEventArgs e) {
            ((XRLabel)sender).Tag = GetCurrentRow();
        }

        private void xrLabel1_PreviewClick(object sender, PreviewMouseEventArgs e) {
            IRow row = (IRow)e.Brick.Value;
            DetailReport detailReport = new DetailReport();
            detailReport.Parameters["catId"].Value = row["CategoryID"];
            detailReport.Parameters["catName"].Value = row["CategoryName"];
            detailReport.ShowPreviewDialog();
        }

        private void xrLabel1_PreviewMouseMove(object sender, PreviewMouseEventArgs e) {
            e.PreviewControl.Cursor = Cursors.Hand;
        }

    }
}

File: VB/MasterReport.vb
Imports System.Data
Imports System.Drawing.Printing
Imports System.Windows.Forms
Imports DevExpress.XtraReports.UI

' ...
Namespace DrillThroughReport

    Public Partial Class MasterReport
        Inherits XtraReport

        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub xrLabel1_BeforePrint(ByVal sender As Object, ByVal e As PrintEventArgs)
            CType(sender, XRLabel).Tag = GetCurrentRow()
        End Sub

        Private Sub xrLabel1_PreviewClick(ByVal sender As Object, ByVal e As PreviewMouseEventArgs)
            Dim detailReport As DetailReport = New DetailReport()
            detailReport.catId.Value = CInt(CType(e.Brick.Value, DataRowView).Row("CategoryID"))
            detailReport.catName.Value = CType(e.Brick.Value, DataRowView).Row("CategoryName").ToString()
            detailReport.ShowPreviewDialog()
        End Sub

        Private Sub xrLabel1_PreviewMouseMove(ByVal sender As Object, ByVal e As PreviewMouseEventArgs)
            e.PreviewControl.Cursor = Cursors.Hand
        End Sub
    End Class
End Namespace
